src/components/invoice/InvoiceComponent.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useState } from 'react';
+import BaseProps from '../../types/BaseProps';
+import { BillingInvoice, BillingInvoiceStatus } from '../../types/Billing';
+import { Text, TouchableOpacity, View } from 'react-native';
+import computeStyleSheet from './InvoiceComponentStyles';
+import computeChipStyleSheet from '../chip/ChipStyle';
+import I18nManager from '../../I18n/I18nManager';
+import I18n from 'i18n-js';
+import Chip from '../chip/Chip';
+import ProviderFactory from '../../provider/ProviderFactory';
+import User from '../../types/User';
+import { DataResult } from '../../types/DataResult';
+import { Card, CardItem, Icon } from 'native-base';
+import Utils from '../../utils/Utils';
+
+export interface Props extends BaseProps {
+  invoice: BillingInvoice;
+}
+
+export default function InvoiceComponent(props: Props) {
```

Comment:
I assume that you had no other choice to use the functional component

Reply:
Go for the class then.
We'll do a full migration to the functional components at once when we will have no other choice.

Reply:
ok